### PR TITLE
Fix config file parsing with multi line values containing quoted parts

### PR DESCRIPTION
**Pull request by csware**
Cross-ref: issue #5459

### CODE REVIEW

File: src/config_parse.c [hunk 1]
```diff
@@ -351,7 +351,7 @@ static int parse_multiline_variable(git_config_parser *reader, git_buf *value, i
 		}
 
 		/* If it was just a comment, pretend it didn't exist */
-		quote_count = strip_comments(line, !!in_quotes);
+		quote_count = strip_comments(line, in_quotes % 2);
```
Comment (pks-t) [inline src/config_parse.c:354]: Is this change even required now? With the below change for how we call `parse_multiline_variable`, were now only ever passing either `0` or `1` to this function, aren't we?

File: tests/resources/config/config12 [hunk 2]
```diff
@@ -11,3 +11,10 @@ and other stuff !@#"
 multiline comment # with ;\n special chars \
 and other stuff !@#"  #^^^
       back = "this is \ba phrase"
+      dollar = some $sign
+      multiquotes = !ls  "x" \
+                    ls  "# comment2" \
+                    $HOME
+      multiquotes2 = !ls  "x" \
+                     ls  "\"# comment2" \
+                     $HOME\"
```
Comment (pks-t) [inline tests/resources/config/config12:20]: Oof. I sometimes really wonder how people come up with such config files :)

Comment (csware) [reply tests/resources/config/config12:20]: Yes, it seems to work without this change.

File: src/config_parse.c (hunk 1 above)
Comment (csware) [reply src/config_parse.c]: Yes, it seems to work without this change.

File: tests/resources/config/config12 (hunk 2 above)
Comment (csware) [reply tests/resources/config/config12:20]: Don't ask ;)

**Review by ethomson — APPROVED** [review]

**Pull request merged**